Unstable tests in TravisCI

I am continuing to frequently get unstable tests in Travis CI despite `@flaky(max_runs=5)`.  Any other suggestions for avoiding these spurious failures?  I'm inclined to increase the tolerances significantly (0.0001 to 0.001).

```
______ BidirectionalAttentionFlowTest.test_model_can_train_save_and_load _______
self = <tests.models.reading_comprehension.bidaf_test.BidirectionalAttentionFlowTest testMethod=test_model_can_train_save_and_load>
    @flaky(max_runs=5)
    def test_model_can_train_save_and_load(self):
>       self.ensure_model_can_train_save_and_load(self.param_file)
tests/models/reading_comprehension/bidaf_test.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
allennlp/common/testing/model_test_case.py:94: in ensure_model_can_train_save_and_load
    name=key)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
field1 = Variable containing:
Columns 0 to 5 
-4.3761e-02 -6.1814e-02 -4.3583e-02 -6.2701e-03  1.0462e-02 -1.6368e-02
-3.9851e... -1.5448e-02  1.0596e-02 -4.2701e-02
-9.2831e-03 -1.5936e-02  9.6128e-03 -4.3546e-02
[torch.FloatTensor of size 5x142]
field2 = Variable containing:
Columns 0 to 5 
-4.3761e-02 -6.1814e-02 -4.3583e-02 -6.2701e-03  1.0462e-02 -1.6368e-02
-3.9851e... -1.5448e-02  1.0596e-02 -4.2701e-02
-9.2831e-03 -1.5936e-02  9.6128e-03 -4.3546e-02
[torch.FloatTensor of size 5x142]

tolerance = 0.0001, name = 'span_end_logits'
    @staticmethod
    def assert_fields_equal(field1, field2, tolerance: float = 1e-6, name: str = None) -> None:
        if isinstance(field1, torch.autograd.Variable):
            assert_allclose(field1.data.numpy(),
                            field2.data.numpy(),
                            rtol=tolerance,
>                           err_msg=name)
E           AssertionError: 
E           Not equal to tolerance rtol=0.0001, atol=0
E           span_end_logits
E           (mismatch 0.1408450704225288%)
E            x: array([[ -4.376138e-02,  -6.181443e-02,  -4.358323e-02,  -6.270148e-03,
E                     1.046205e-02,  -1.636827e-02,  -2.302323e-02,   2.581067e-03,
E                    -1.454758e-02,  -1.829709e-02,  -2.281654e-02,  -2.857263e-02,...
E            y: array([[ -4.376138e-02,  -6.181443e-02,  -4.358323e-02,  -6.270148e-03,
E                     1.046205e-02,  -1.636827e-02,  -2.302323e-02,   2.581067e-03,
E                    -1.454758e-02,  -1.829709e-02,  -2.281654e-02,  -2.857263e-02,...
allennlp/common/testing/model_test_case.py:104: AssertionError
```

--- a/allennlp/common/testing/model_test_case.py
+++ b/allennlp/common/testing/model_test_case.py
@@ -29,7 +29,7 @@ class ModelTestCase(AllenNlpTestCase):
         self.dataset = dataset
         self.model = Model.from_params(self.vocab, params['model'])
 
-    def ensure_model_can_train_save_and_load(self, param_file: str):
+    def ensure_model_can_train_save_and_load(self, param_file: str, tolerance: int = 1e-6):
         save_dir = os.path.join(self.TEST_DIR, "save_and_load_test")
         archive_file = os.path.join(save_dir, "model.tar.gz")
         model = train_model_from_file(param_file, save_dir)
@@ -64,7 +64,7 @@ class ModelTestCase(AllenNlpTestCase):
                 for subfield in field:
                     self.assert_fields_equal(model_batch[key][subfield],
                                              loaded_batch[key][subfield],
-                                             tolerance=1e-6,
+                                             tolerance=tolerance,
                                              name=key + '.' + subfield)
             else:
                 self.assert_fields_equal(model_batch[key], loaded_batch[key], 1e-6, key)

--- a/allennlp/common/testing/model_test_case.py
+++ b/allennlp/common/testing/model_test_case.py
@@ -29,7 +29,7 @@ class ModelTestCase(AllenNlpTestCase):
         self.dataset = dataset
         self.model = Model.from_params(self.vocab, params['model'])
 
-    def ensure_model_can_train_save_and_load(self, param_file: str, tolerance: int = 1e-6):
+    def ensure_model_can_train_save_and_load(self, param_file: str, tolerance: float = 1e-6):
         save_dir = os.path.join(self.TEST_DIR, "save_and_load_test")
         archive_file = os.path.join(save_dir, "model.tar.gz")
         model = train_model_from_file(param_file, save_dir)

--- a/tests/models/reading_comprehension/bidaf_test.py
+++ b/tests/models/reading_comprehension/bidaf_test.py
@@ -46,7 +46,7 @@ class BidirectionalAttentionFlowTest(ModelTestCase):
     # `masked_softmax`...) have made this _very_ flaky...
     @flaky(max_runs=5)
     def test_model_can_train_save_and_load(self):
-        self.ensure_model_can_train_save_and_load(self.param_file)
+        self.ensure_model_can_train_save_and_load(self.param_file, 2e-5)
 
     @flaky
     def test_batch_predictions_are_consistent(self):

--- a/tests/models/reading_comprehension/bidaf_test.py
+++ b/tests/models/reading_comprehension/bidaf_test.py
@@ -46,7 +46,7 @@ class BidirectionalAttentionFlowTest(ModelTestCase):
     # `masked_softmax`...) have made this _very_ flaky...
     @flaky(max_runs=5)
     def test_model_can_train_save_and_load(self):
-        self.ensure_model_can_train_save_and_load(self.param_file, 2e-5)
+        self.ensure_model_can_train_save_and_load(self.param_file, tolerance=2e-5)
 
     @flaky
     def test_batch_predictions_are_consistent(self):

